react/Search.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { FC, Fragment } from 'react'
+import { prop } from 'ramda'
+import { useQuery } from 'react-apollo'
+import RightArrow from './components/icons/RightArrow'
+import searchEngine from './graphql/search.graphql'
+
+interface Props {
+  query: {
+    q: string
+  }
+}
+
+interface SearchResult {
+  title: string
+  snippet: string
+  link: string
+}
+
+const Search: FC<Props> = ({ query }) => {
+  const queryString = query.q || ''
+
+  const { data } = useQuery(searchEngine, {
+    variables: { searchQuery: queryString },
+  })
+
+  const results: [SearchResult] = prop('searchEngine', data) || []
+
+  const isNotLastResult = (results: [SearchResult], index: number) =>
+    index !== results.length - 1
+
+  return (
```

Comment:
One small detail, we might need a loading state, so that the results page doesn't "flash" with "No results found" then the actual results

Reply:
Good point!